Add Python3 example (#2)
upvestco/http-signature-examples: Python/README.md

# Overview

This directory contains the basic outline of a working Upvest
Investment API client in Python3.

The intent of this code is to show you how to implement V15 of the
HTTP Message Signing mechanism and authorisation token acquisation
you'll need to securely access the Upvest Investment API.

# Installing the example

**Prerequisites**

> - It is assumed that you have python 3 installed on your machine,

Reviewer: ```suggestion
- It is assumed that you have Python3 installed on your machine,
```

  following whatever procedure is suggested/validated in your company.

> The easiest way to prepare your machine to the run the example is to

Reviewer: ```suggestion
The easiest way to prepare your machine to run the example is to
```

> use a Python3 virtual environment. The following steps will clone this

Reviewer: ```suggestion
use a Python 3 virtual environment. The following steps will clone this
```

repository and set up such an environment:

```sh
git clone git@github.com/toknapp/http-signature-examples.git 
cd http-signature-examples/Python
python3 -m venv ./venv
source ./venv/bin/activate
```

You should now be in a Python3 virtual environment, as indicated by
your shell prompt, which should now look something like this:

```sh
(venv) your-name@your-computer Python % 
```

Should you wish to exit the virtual environment, you can simply issue
the `deactivate` command.

> You can now install the projects dependencies into your virtual

Reviewer: ```suggestion
You can now install the project's dependencies into your virtual
```

environment (remembering to reactivate it first if you just issued the
`deactivate` command).

```sh
pip3 install -r requirements.txt
```

This single command will ensure that all project dependencies are in place.

# Running the example
You should be able to run the `sign.py` script in this directory by
invoking the following (replacing the values with your real PEM file,
PEM Password, pre-shared Key ID, Client ID and Client Secret):

```sh
python3 sign.py "./my_ecdsa521.pem" "my-secret-pem-password" "pre-shared-key-id" \
	"my-client-id" "my-client-secret"
```

# Understanding the example

Please make sure you have read and understood the associated
[documentation on implementing message

> signing.](https://docs.upvest.co/tutorials/implementing_http_signatures)

Reviewer: Since this code example is implementing v15, we can directly link there, now that the v15 docs are online.

```suggestion
signing.](https://docs.upvest.co/tutorials/implementing_http_signatures_v15)
```
